code/script.js

```diff
@@ -1,51 +1,221 @@
+  chat.scrollTop = chat.scrollHeight;
+}
+
+//Functions for bot response and user response
+const botResponse = (msg) => {
+  showMessage(msg, "bot");
 }
 
-// Starts here
+const userResponse = (msg) => {
+  showMessage(msg, "user");
+}
+
+//------------QUESTION 1-------------
+// Greeting function from the bot
 const greetUser = () => {
-  // here we call the function showMessage, that we declared earlier with the argument "Hello there, What's your name?" for message, and the argument "bot" for sender
-  showMessage("Hello there, What's your name?", 'bot')
-  // Just to check it out, change 'bot' to 'user' here 👆
+  // call the function for bot response the argument "Hello there! What's your name?" 
+  botResponse("Hello there! What's your name?");
+}
+
+// When website loaded, chatbot asks first question. Greeting function will be called one second after the website is loaded.
+setTimeout(greetUser, 1000);
+
+// Add event listener to input form to handle name input
+inputForm.addEventListener("submit", (event) => {
+  // Prevents the form from actually submitting and refreshing the page
+  event.preventDefault();
+  
+  // Get user input
+  const userName = nameInput.value;
+
+  // Display the user's name
+  userResponse(`Hello! I am ${userName}.`);
+
+  // Clear input field after a message has been sent
+  nameInput.value = "";
+  
+  // After one second, show the next question about flavours by invoking the next function, passing the user's name into it so it appears in the next question from the bot
+  setTimeout(() => questionAboutFlavour(userName), 1000);
+})
+
+//------------QUESTION 2-------------
+// Bot shows flavour options upon receiving user's name
+const questionAboutFlavour = (userName) => {
+  botResponse(`Time to treat yourself some icecream, ${userName}. Which flavour would you like to test today?`);
+
+  // Remove the input field and send button
+  inputForm.remove();  
+  
+  // Display buttons for flavours
+  inputWrapper.innerHTML += `
+    <button id="chocolate" value="Chocolate flavour">Chocolate</button>
+    <button id="strawberry" value="Strawberry flavour">Strawberry</button>
+    <button id="vanilla" value="Vanilla flavour">Vanilla</button>
+    `;
+  
+  // Store the buttons in variables
+  const chocolateButton = document.getElementById("chocolate");
+  const strawberryButton = document.getElementById("strawberry");
+  const vanillaButton = document.getElementById("vanilla");
+
+  // Function to clear the buttons and move to the next question
+  const moveToScoopQuestion = () => {
+    chocolateButton.remove();
+    strawberryButton.remove();
+    vanillaButton.remove();
+    setTimeout(() => questionAboutNumberOfScoops(), 1000);
+  };
+
+  // Function to add event listener to click on buttons so user's choice is displayed and bot shows confirmation message
+  const flavourConfirmation = (button) => {
+    button.addEventListener("click", () => {
+      userResponse(`I would love to have ${button.value}!`);
```

Comment:
you can store the value somewhere in the code so you can. use it later when you want to calculate the price or show it later this value with all of the options